blockchain/blockindex.go

```diff
@@ -150,6 +153,27 @@ func (node *blockNode) Header() wire.BlockHeader {
 	}
 }
 
+// Turn the lowest '1' bit in the binary representation of a number into a '0'
+func invertLowestOne(n int32) int32 {
+	return n & (n - 1)
+}
+
+// Compute what height to jump back to during blocknode.Ancestor()
+func getSkipHeight(height int32) int32 {
+	if height < 2 {
+		return 0
+	}
+
+	// Determine which height to jump back to. Any number strictly lower than height is acceptable,
+	// but the following expression seems to perform well in simulations (max 110 steps to go back
+	// up to 2**18 blocks).
+	if (height & 1) == 1 {
+		return invertLowestOne(invertLowestOne(height-1)) + 1
+	} else {
+		return invertLowestOne(height)
```

Comment:
Isn't this just a noop if the LSB is zero? Why not just return `height` here directly.

Reply:
Changed the logic